### DISCUSSION TITLE
Literacy as a skill

### DISCUSSION THREAD

**Discussion by scarf005** [cataclysmbn/Cataclysm-BN#2072] [General]
### Is your feature request related to a problem? Please describe.

`Iliterate` flag permanently disables ability to read, even if player is Einstein reincarnation with INT 20. However, it is is possible for iliterate adults to learn how to read or write .

#### Literacy skill
Add a new 'hidden skill', `literacy`. Below is an example of how they could work like:

- Iliterate starts with 0.
- Kids start with 4.
- Adults start with 5 (out of 10).

Literacy levels determine your ability to read and speed boost on reading.
- Levels from 2 to 4 can only recognize some of the words of a text. descriptions would be gibberish  like: 
```
(*$#(
S kah minutes 8*(%#
v 9&*A*SFA% and (...)
```
- Levels from 4 can read, but with penalties
- Levels from 6 can read books without penalty. 
- Higher literacy level boosts reading time

### Increasing skill level
- Levels below 3 could only increase literacy by getting taught from NPCs.
- Reading books (being read by NPCs count) increase skill.
- Int 8 iliterate player would be able to reach literacy level 4 after constantly learning from NPCs for a season or two.

### Additional context

https://www.quora.com/Is-it-possible-for-an-illiterate-person-to-learn-reading-and-writing-on-his-or-her-own

**Comment by Coolthulhu** [comment]
I don't see the benefit to gameplay here.
Illiterate trait is supposed to be a gameplay limitation, not something that you can just grind away.

If anything, I'd rather disable the "NPC reads to you", since it makes the trait not do its job if you have at least one NPC sitting in your crafting base.

**Reply by leoCottret** [reply]
I think having someone being able to read you the book makes sense, although at a lower rate if that's not the case already.
Also scarf is right, illiterate adults can learn to read, even if it's harder for them.
I think it even stays hard with time, but maybe I'm mistaken.
Also I'm all for balance fixes but they need but at least be somewhat adjusted on reality (thinking of changes like the near sighted/far sighted restrictions).

**Comment by RodriguesCIA** [comment]
> I don't see the benefit to gameplay here. Illiterate trait is supposed to be a gameplay limitation, not something that you can just grind away.
> 
> If anything, I'd rather disable the "NPC reads to you", since it makes the trait not do its job if you have at least one NPC sitting in your crafting base.

That said, isn't overcoming your inherent weaknesses part of progression? Is a churl going to, forever, have to grind everything by crafting ad infinitum when they could get some friendly npc to do the nerd shit for them? Also, NPCs are, by default, pretty rare to get.

**Comment by scarf005** [comment]
that said, not having ILiterate trait could be the simpler solution.
